webdriver/client.py

```diff
@@ -349,10 +350,14 @@ def send_raw_command(self, method, url, body=None, headers=None):
         :return: an instance of wdclient.Response describing the HTTP response
             received from the remote end
         """
-        url = urlparse.urljoin("session/%s/" % self.session_id, url)
-        return self.transport.send(method, url, body, headers)
+        response = self.transport.send(method, url, body, headers)
+        if response.status != 200:
+            value = response.body["value"]
+            cls = error.get(value.get("error"))
+            raise cls(value.get("message"))
```

Comment:
I'd strongly prefer to move this exception-raising code from `send_raw_request`
to `send_request`. My intention with `send_raw_request` was to define a
low-level method for fine-grained testing of the protocol, but this feature
tends to blur the "low-level"/"high-level" distinction. It means that tests for
expected protocol errors will need to be expressed in terms of Python
exceptions, something I very much want to avoid (since they are not
"exceptional" in those cases and since readers may have a hard time relating
those runtime exceptions back to the protocol itself).

My apologies for the late response here. Given that we decided the main intent
of this patch was mistaken, I wasn't expecting it to land.